### ISSUE TITLE
[BUG] ActionListOpsWidget breaks with input titles starting with a dash

### ISSUE THREAD

**Issue by Jermolene** [Jermolene/TiddlyWiki5#6535]
The `<$action-listops>` widget breaks with input titles starting with a dash. For example, if the tiddler "target" contains the text `-3` then the following will not work as expected:

```
<$action-listops  $tiddler="target" $field="text" $subfilter="+[toggle[-3]]"/>
```

In this case, the text field of "target" is not left untouched, and the "-3" item is not removed.

But, the same experiment with "foobar" instead of "-3" works as expected.

The problem is here:

https://github.com/Jermolene/TiddlyWiki5/blob/c741978e959c2608fa01dcd881d47194ccdf79b8/core/modules/widgets/action-listops.js#L72-L76

The code makes the assumption that a stringified list can be evaluated as a filter and will get the same results. That assumption fails if the list item starts with a dash, as it is interpreted as the `:except` filter run prefix.

**Comment by twMat** [comment]
I have no idea if it is the same problem but titles starting with a `+` also seem to misbehave. If the target tiddler is blank and you toggle it once it works, but clicking again adds a second copy of the operand but without the `+`.

**Comment by ericshulman** [comment]
The same problem happens for titles that start with **any** symbol that can start a filter run: `+ - = ~ : `
It should occur for `[` and `]`, but these have to be contained in variable operands, since they can't be used as literal operand values because of the enclosing square brackets used by the filter operator syntax; i.e., `$subfilter="+[toggle<var>]"`

**Comment by Jermolene** [comment]
We've already explored one potential fix: updating `$tw.utils.stringifyList()` to ensure that it add double square bracket quotes around titles that start with problematic characters (we would also presumably want to wrap blank values, which are currently ignored – see #1451)

* Ticket #4084 discusses the problem
* PR #4087 was merged with the changes to stringifyList
* #4248 discusses a potential problem with the fix that we decided was actually the intended behaivour
* PR #4249 reverts the changes in #4087

On the basis of those previous discussions, I think we need to explore other solutions. In particular, we might be able to provide the input list as the source for the filter evaluation, and prefix the subfilter with `[all[]] ` for evaluation to get the titles back.

**Comment by Jermolene** [comment]
This is now fixed in #6540, let me know how you get on @ericshulman
